From ef3f8c33c24a2f7255eb03487e768040d69f5976 Mon Sep 17 00:00:00 2001
From: Xulin Sun <xulin.sun@windriver.com>
Date: Wed, 8 Mar 2017 15:01:22 +0800
Subject: [PATCH] phy: phy-bcm-ns2-usbdrd: replace usleep_range with udelay

To fix the below calltrace:

BUG: scheduling while atomic: swapper/0/1/0x00000002
Modules linked in:
CPU: 2 PID: 1 Comm: swapper/0 Not tainted 4.8.18-WR9.0.0.4_standard #1
Hardware name: Broadcom NS2 SVK (DT)
Call trace:
[<ffffff8008089ce8>] dump_backtrace+0x0/0x1d8
[<ffffff8008089ee4>] show_stack+0x24/0x30
[<ffffff800844dd6c>] dump_stack+0x94/0xb8
[<ffffff80080c60c4>] __schedule_bug+0x64/0x80
[<ffffff80088177d4>] __schedule+0x4dc/0x628
[<ffffff8008817964>] schedule+0x44/0xb0
[<ffffff800881b1d0>] schedule_hrtimeout_range_clock+0x98/0x110
[<ffffff800881b280>] schedule_hrtimeout_range+0x38/0x48
[<ffffff800881af80>] usleep_range+0x60/0x70
[<ffffff8008484c04>] ns2_drd_phy_poweron+0x1dc/0x268
[<ffffff8008481d60>] phy_power_on+0x78/0x100
[<ffffff80085daefc>] ehci_platform_power_on+0xcc/0x178
[<ffffff80085db120>] ehci_platform_probe+0x178/0x520
[<ffffff8008502e40>] platform_drv_probe+0x60/0xc8
[<ffffff8008500c14>] driver_probe_device+0x204/0x2b0
[<ffffff8008500d7c>] __driver_attach+0xbc/0xc0
[<ffffff80084fe8b8>] bus_for_each_dev+0x70/0xb0
[<ffffff80085002d0>] driver_attach+0x30/0x40
[<ffffff80084ffdd0>] bus_add_driver+0x118/0x238
[<ffffff80085018b8>] driver_register+0x68/0x100
[<ffffff8008502d64>] __platform_driver_register+0x54/0x60
[<ffffff8008a88ec0>] ehci_platform_init+0x50/0x60
[<ffffff8008083ae4>] do_one_initcall+0x44/0x138
[<ffffff8008a50d08>] kernel_init_freeable+0x1a0/0x274
[<ffffff80088163c0>] kernel_init+0x18/0x108

In an atomic operation, sleep would cause above calltrace in the initialization
of the system, just use udelay to replace.

Signed-off-by: Xulin Sun <xulin.sun@windriver.com>
---
 drivers/phy/phy-bcm-ns2-usbdrd.c |    2 +-
 1 files changed, 1 insertions(+), 1 deletions(-)

diff --git a/drivers/phy/phy-bcm-ns2-usbdrd.c b/drivers/phy/phy-bcm-ns2-usbdrd.c
index 4c045ec..aa6cdf1 100644
--- a/drivers/phy/phy-bcm-ns2-usbdrd.c
+++ b/drivers/phy/phy-bcm-ns2-usbdrd.c
@@ -261,7 +261,7 @@ static int ns2_drd_phy_poweron(struct phy *phy)
 		writel(val, driver->idmdrd_rst_ctrl);
 
 		/* Give hardware time to settle */
-		usleep_range(150, 300);
+		udelay(300);
 
 		/* port over current Polarity */
 		val = readl(driver->usb2h_strap_reg);
-- 
1.7.5.4

